src/microtar.c

```diff
@@ -97,19 +111,21 @@ static int raw_to_header(mtar_header_t *h, const mtar_raw_header_t *rh) {
 
   /* Build and compare checksum */
   chksum1 = checksum(rh);
-  sscanf(rh->checksum, "%o", &chksum2);
+  sscanf(rh->checksum, "%8o", &chksum2);
   if (chksum1 != chksum2) {
     return MTAR_EBADCHKSUM;
   }
 
   /* Load raw header into header */
-  sscanf(rh->mode, "%o", &h->mode);
-  sscanf(rh->owner, "%o", &h->owner);
-  sscanf(rh->size, "%o", &h->size);
-  sscanf(rh->mtime, "%o", &h->mtime);
+  sscanf(rh->mode, "%8o", &h->mode);
+  sscanf(rh->owner, "%8o", &h->owner);
+  sscanf(rh->size, "%12o", &h->size);
+  sscanf(rh->mtime, "%12o", &h->mtime);
   h->type = rh->type;
-  strcpy(h->name, rh->name);
-  strcpy(h->linkname, rh->linkname);
+  
+  // Here we can memcpy because both buffers have the same size
+  memcpy(h->name, rh->name, name_buf_width);
```

Comment:
Use sizeof for array here for either h->name or rh->name. They both are same sized.